lib/utils/findPort.js

```diff
@@ -1,32 +1,6 @@
 'use strict';
 
-const pRetry = require('p-retry');
-const portfinder = require('portfinder');
+// TODO(@anshumanv) - Drop in next major release
+const getFreePort = require('../Server').getFreePort;
 
-function runPortFinder() {
-  return new Promise((resolve, reject) => {
-    // default port
-    portfinder.basePort = 8080;
-    portfinder.getPort((error, port) => {
-      if (error) {
-        return reject(error);
-      }
-
-      return resolve(port);
-    });
-  });
-}
-
-function findPort(port) {
-  if (port && port !== 'auto') {
-    return Promise.resolve(port);
-  }
-
-  // Try to find unused port and listen on it for 3 times,
-  // if port is not specified in options.
-  const defaultPortRetry = parseInt(process.env.DEFAULT_PORT_RETRY, 10) || 3;
-
-  return pRetry(runPortFinder, { retries: defaultPortRetry });
-}
-
-module.exports = findPort;
+module.exports = getFreePort;
```

Comment:
Can we add simple tests for this and put them in `Server.test.api` too, we will remove it in future, so it is fine place

Reply:
oh yep lets do it

Reply:
done